pkg/apis/rke.cattle.io/v1/machine.go

```diff
@@ -8,21 +8,62 @@ import (
 )
 
 type RKECommonNodeConfig struct {
-	Labels                    map[string]string `json:"labels,omitempty"`
-	Taints                    []corev1.Taint    `json:"taints,omitempty"`
-	CloudCredentialSecretName string            `json:"cloudCredentialSecretName,omitempty"`
+	//+optional
+	//+nullable
+	Labels map[string]string `json:"labels,omitempty"`
+
+	//+optional
+	//+nullable
+	Taints []corev1.Taint `json:"taints,omitempty"`
+
+	//+optional
+	//+nullable
+	CloudCredentialSecretName string `json:"cloudCredentialSecretName,omitempty"`
 }
 
 type RKEMachineStatus struct {
-	Conditions                []genericcondition.GenericCondition `json:"conditions,omitempty"`
-	JobName                   string                              `json:"jobName,omitempty"`
-	Ready                     bool                                `json:"ready,omitempty"`
-	DriverHash                string                              `json:"driverHash,omitempty"`
-	DriverURL                 string                              `json:"driverUrl,omitempty"`
-	CloudCredentialSecretName string                              `json:"cloudCredentialSecretName,omitempty"`
-	FailureReason             string                              `json:"failureReason,omitempty"`
-	FailureMessage            string                              `json:"failureMessage,omitempty"`
-	Addresses                 []capi.MachineAddress               `json:"addresses,omitempty"`
+	//+optional
+	//+nullable
+	// Conditions is a representation of the machine's current state.
+	Conditions []genericcondition.GenericCondition `json:"conditions,omitempty"`
+
+	//+optional
+	//+nullable
+	// JobName is the name of the provisioning job of the machine.
+	JobName string `json:"jobName,omitempty"`
+
+	// Ready indicates whether the provider ID has been set in this machine's spec.
```

Comment:
Note to reviewer: This is about the only thing I could figure out from the code for this field. Does it have another meaning?